src/domain/challenge/challenge/challenge.resolver.mutations.ts

```diff
@@ -91,10 +97,24 @@ export class ChallengeResolverMutations {
       opportunityData,
       agentInfo
     );
-    return await this.opportunityAuthorizationService.applyAuthorizationPolicy(
+    await this.opportunityAuthorizationService.applyAuthorizationPolicy(
       opportunity,
       challenge.authorization
     );
+
+    const opportunityCreatedEvent: OpportunityCreatedPayload = {
+      eventID: `challenge-opportunity-created-${Math.round(
+        Math.random() * 100
+      )}`,
+      challengeID: challenge.id,
+      opportunity,
+    };
+    await this.opportunityCreatedSubscription.publish(
```

Comment:
i don't think you need to wait for this

Reply:
Well, no I don't. I don't depend on whether the created challenge is published.